internal/pkg/propagation/summaries.go

```diff
@@ -0,0 +1,761 @@
+	"errors.New": fromFirstArgToFirstRet,
+	// func Unwrap(err error) error {
+	"errors.Unwrap": fromFirstArgToFirstRet,
+	// func As(err error, target interface{}) bool {
+	"errors.As": {
+		ifTainted:   0b1,
+		taintedArgs: []int{1},
+	},
+	// func SplitN(s, sep string, n int) []string {
+	"strings.SplitN": fromFirstArgToFirstRet,
+	// func SplitAfterN(s, sep string, n int) []string {
+	"strings.SplitAfterN": fromFirstArgToFirstRet,
+	// func Split(s, sep string) []string {
+	"strings.Split": fromFirstArgToFirstRet,
+	// func SplitAfter(s, sep string) []string {
+	"strings.SplitAfter": fromFirstArgToFirstRet,
+	// func Fields(s string) []string {
+	"strings.Fields": fromFirstArgToFirstRet,
+	// func FieldsFunc(s string, f func(rune) bool) []string {
+	"strings.FieldsFunc": fromFirstArgToFirstRet,
+	// func Join(elems []string, sep string) string {
+	"strings.Join": {
+		ifTainted:   0b11,
+		taintedRets: []int{0},
+	},
+	// func Map(mapping func(rune) rune, s string) string {
+	"strings.Map": {
+		ifTainted:   0b10,
+		taintedRets: []int{0},
+	},
+	// func Repeat(s string, count int) string {
+	"strings.Repeat": fromFirstArgToFirstRet,
+	// func ToUpper(s string) string {
+	"strings.ToUpper": fromFirstArgToFirstRet,
+	// func ToLower(s string) string {
+	"strings.ToLower": fromFirstArgToFirstRet,
+	// func ToTitle(s string) string {
+	"strings.ToTitle": fromFirstArgToFirstRet,
+	// func ToUpperSpecial(c unicode.SpecialCase, s string) string {
+	"strings.ToUpperSpecial": {
+		ifTainted:   0b10,
+		taintedRets: []int{0},
+	},
+	// func ToLowerSpecial(c unicode.SpecialCase, s string) string {
+	"strings.ToLowerSpecial": {
+		ifTainted:   0b10,
+		taintedRets: []int{0},
+	},
+	// func ToTitleSpecial(c unicode.SpecialCase, s string) string {
+	"strings.ToTitleSpecial": {
+		ifTainted:   0b10,
+		taintedRets: []int{0},
+	},
+	// func ToValidUTF8(s, replacement string) string {
+	"strings.ToValidUTF8": {
+		ifTainted:   0b11,
+		taintedRets: []int{0},
+	},
+	// func Title(s string) string {
+	"strings.Title": fromFirstArgToFirstRet,
+	// func TrimLeftFunc(s string, f func(rune) bool) string {
+	"strings.TrimLeftFunc": fromFirstArgToFirstRet,
+	// func TrimRightFunc(s string, f func(rune) bool) string {
+	"strings.TrimRightFunc": fromFirstArgToFirstRet,
+	// func TrimFunc(s string, f func(rune) bool) string {
+	"strings.TrimFunc": fromFirstArgToFirstRet,
+	// func Trim(s, cutset string) string {
+	"strings.Trim": fromFirstArgToFirstRet,
+	// func TrimLeft(s, cutset string) string {
+	"strings.TrimLeft": fromFirstArgToFirstRet,
+	// func TrimRight(s, cutset string) string {
+	"strings.TrimRight": fromFirstArgToFirstRet,
+	// func TrimSpace(s string) string {
+	"strings.TrimSpace": fromFirstArgToFirstRet,
+	// func TrimPrefix(s, prefix string) string {
+	"strings.TrimPrefix": fromFirstArgToFirstRet,
+	// func TrimSuffix(s, suffix string) string {
+	"strings.TrimSuffix": fromFirstArgToFirstRet,
+	// func Replace(s, old, new string, n int) string {
```

Comment:
This is a curious case.  I could see someone using `Replace` to specifically purge a secret, e.g. `Replace(problematic, secret, "REDACTED", -1)`.  Probably not an issue until someone hits it, but we might want to make a note.

Ditto `ReplaceAll`.

Reply:
Yeah, some of these are a bit weird. When in doubt I leaned towards including a function rather than excluding it.

Since we check for sanitizers before propagating, I think this should be fine, i.e. if a user really wants to use string replacements to perform sanitization, they could configure `Replace*` to be sanitizers.

RE `we might want to make a note`: would you like me to include the above reasoning in a comment somewhere?

Reply:
Good points.  And yeah, I don't think there's really a good place to make a note of it that is somewhere we would remember to reference.  The Note will be if someone opens an Issue for it, then.